### PR TITLE
Refactor Makefile path handling

### PR DESCRIPTION
**Pull request by kbara**
Pflua uses recursive make, without autotools.
This change adds a $ABS_TOP_SRCDIR variable, inspired by autoconf,
to make the way the Makefiles deal with paths be less fragile.

Note that this intentionally means that only the top-level Makefile should
be invoked by hand; the rest now depend on it. For now, this can be worked
around by setting ABS_TOP_SRCDIR appropriately while invoking the dependent
Makefiles, but this is not advised.

### CODE REVIEW

File: doc/Makefile [hunk 1]
```diff
@@ -1,4 +1,6 @@
-PATH := ../deps/luajit/usr/local/bin:$(PATH)
+TOP_SRCDIR=../
```
Comment (andywingo) [inline doc/Makefile]: nit: no trailing slash on directory variables

File: Makefile [hunk 2]
```diff
@@ -1,4 +1,5 @@
-LUAJIT_O := deps/luajit/src/libluajit.a
+ABS_TOP_SRCDIR=`pwd`
```
Comment (andywingo) [inline Makefile]: Could we do this in all files:

TOP_SRCDIR:=<whatever>
include $(TOP_SRCDIR)/common.mk

and in common.mk

ABS_TOP_SRCDIR:=$(shell cd $(TOP_SRCDIR) && pwd)

Additionally in common.mk we could define LUAJIT as the full path to luajit, and that will eventually let us remove all PATH manipulations in our makefiles as we can just reference binaries by their absolute paths using convenient make variables.  WDYT?

**Comment by andywingo** [comment]
LGTM if travis passes.  Thank you!

**Comment by kbara** [comment]
Merged, thank you and you're welcome.

**Pull request merged**